code/lib/cli/src/generate.ts

```diff
@@ -19,309 +20,323 @@ import { withTelemetry } from '@storybook/core/core-server';
 import type { CommandOptions } from './generators/types';
 import { initiate } from './initiate';
 import { add } from './add';
-import { migrate } from './migrate';
 import { upgrade, type UpgradeOptions } from './upgrade';
 import { sandbox } from './sandbox';
 import { link } from './link';
-import { doAutomigrate } from './automigrate';
 import { dev } from './dev';
 import { build } from './build';
 import { doctor } from './doctor';
 
-addToGlobalContext('cliVersion', versions.storybook);
+const args = process.argv.slice(2);
 
-const pkg = findPackageSync(__dirname);
-invariant(pkg, 'Failed to find the closest package.json file.');
-const consoleLogger = console;
+// Forward some commands to @storybook/toolbox
+if (['migrate', 'automigrate'].includes(args[0])) {
+  const command = ['npx', `@storybook/toolbox@${versions.storybook}`, ...args];
+  console.log(command.join(' '));
+  const child = spawn(command[0], command.slice(1), { stdio: 'inherit' });
+  child.on('exit', (code) => {
+    if (code != null) {
+      process.exit(code);
+    }
+    process.exit(1);
+  });
+} else {
```

Comment:
would love being able to do an early return here, but is not possible with ESM modules.